core/src/main/java/org/ethereum/beacon/core/state/ValidatorRecord.java

```diff
@@ -258,6 +260,11 @@ public Builder withStatusFlags(UInt64 statusFlags) {
       return this;
     }
 
+    public Builder withStatusFlags(Function<UInt64, UInt64> statusFlagsUpdater) {
```

Comment:
Why not just update flags with pre-calculated value?

Reply:
Because we need to *modify* existing flag:
```python
     validator.status_flags |= INITIATED_EXIT
```

Reply:
```java
validtor.withStatusFlags(validator.getStatusFlags().or(INITIATED_EXIT));
```
looks pretty similar to
```java
validtor.withStatusFlags(flags -> flags.or(INITIATED_EXIT));
```

Anyway, not a big deal

Reply:
This is `Builder` instance passed to this method 
```
withValidatorRecord(ValidatorIndex idx,
      Consumer<ValidatorRecord.Builder> validatorUpdater)
```
it has no `getStatusFlags()`
Anyway, not a big deal :)

Reply:
Oh, got it!